### PR TITLE
Feature/reef support

### PR DESCRIPTION
**Pull request by bnsports**

### CODE REVIEW

File: fearless/Common/Services/RemoteSubscription/StakingAccountSubscription/StakingAccountSubscriptionV14.swift [hunk 1]
```diff
@@ -0,0 +1,147 @@
+import Foundation
+import RobinHood
+import IrohaCrypto
+import SSFUtils
+import SSFModels
+
+final class StakingAccountSubscriptionV14: BaseStakingAccountSubscription {
+    override func subscribeRemote(for accountId: AccountId) {
+        mutex.lock()
```
Comment (DRadmir) [inline fearless/Common/Services/RemoteSubscription/StakingAccountSubscription/StakingAccountSubscriptionV14.swift:9]: please check, perhaps you can use super method?

File: fearless/Common/Services/RemoteSubscription/StakingAccountSubscription/StakingAccountSubscriptionV14.swift [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+            operationQueue.addOperations(operations, waitUntilFinished: false)
+
+        } catch {
+            logger?.error("Did receive unexpected error \(error)")
+        }
+    }
+
+    override func subscribeRemote(for stashItem: StashItem) {
+        mutex.lock()
```
Comment (DRadmir) [inline fearless/Common/Services/RemoteSubscription/StakingAccountSubscription/StakingAccountSubscriptionV14.swift:79]: please check, perhaps you can use super method?

File: fearless/Common/Substrate/CallFactory/SubstrateCallFactoryDefault.swift [hunk 3]
```diff
@@ -625,7 +625,7 @@ class SubstrateCallFactoryDefault: SubstrateCallFactoryProtocol {
         to receiver: AccountId,
         amount: BigUInt
     ) -> any RuntimeCallable {
-        let args = TransferCall(dest: .accoundId(receiver), value: amount, currencyId: nil)
+        let args = TransferCall(dest: .indexedString(receiver), value: amount, currencyId: nil)
```
Comment (DRadmir) [inline fearless/Common/Substrate/CallFactory/SubstrateCallFactoryDefault.swift:628]: Does it work successfully with 14 metadata?

File: fearless/Common/Substrate/Calls/BondCall.swift [hunk 4]
```diff
@@ -3,9 +3,38 @@ import SSFUtils
 import BigInt
 
 struct BondCall: Codable {
+    enum CodingKeys: String, CodingKey {
```
Comment (DRadmir) [inline fearless/Common/Substrate/Calls/BondCall.swift:6]: all changes below in this file compatible with 13 and 14 metadata?

File: fearless/Common/Substrate/Types/AccountInfo.swift [hunk 5]
```diff
@@ -268,3 +268,21 @@ enum EquilibruimPositive: Decodable {
         }
     }
 }
+
+extension AccountInfo: ScaleDecodable {
```
Comment (DRadmir) [inline fearless/Common/Substrate/Types/AccountInfo.swift:272]: please check, needed this extensions?

File: fearless/Common/Substrate/Types/Nomination.swift [hunk 6]
```diff
@@ -2,8 +2,26 @@ import Foundation
...
+
     let targets: [Data]
     @StringCodable var submittedIn: UInt32
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        do {
+            targets = try container.decode([Data].self, forKey: .targets)
+        } catch {
```
Comment (DRadmir) [inline fearless/Common/Substrate/Types/Nomination.swift:18]: nice

File: fearless/Common/Substrate/Types/SuperIdentity.swift [hunk 7]
```diff
@@ -16,8 +16,14 @@ struct SuperIdentity: Codable {
...
 
-        parentAccountId = try container.decode(Data.self)
+        do {
+            parentAccountId = try container.decode(Data.self)
+        } catch {
+            let parentAccountIdString = try container.decode(String.self)
+            parentAccountId = Data(hex: parentAccountIdString)
+        }
         data = try container.decode(ChainData.self)
+        print("success")
```
Comment (DRadmir) [inline fearless/Common/Substrate/Types/SuperIdentity.swift:26]: without this print will not work

File: fearless/CoreLayer/OperationFactory/BlockExplorer/History/Main/ReefSubsquidHistoryOperationFactory.swift [hunk 8]
```diff
@@ -0,0 +1,378 @@
...
+        """
+    }
+
+    private func prepareFilter(
+        filters: [WalletTransactionHistoryFilter],
+        address: String
+    ) -> String {
+        var filterStrings: [String] = []
+
+//        if filters.contains(where: { $0.type == .other && $0.selected }) {
```
Comment (DRadmir) [inline fearless/CoreLayer/OperationFactory/BlockExplorer/History/Main/ReefSubsquidHistoryOperationFactory.swift:99]: comments

File: fearless/CoreLayer/OperationFactory/BlockExplorer/History/Staking/ParachainHistoryOperationFactory.swift [hunk 9]
```diff
@@ -22,7 +22,7 @@ enum ParachainHistoryOperationFactoryAssembly {
             return ParachainSubsquidHistoryOperationFactory(url: blockExplorer?.url)
         case .sora:
             return ParachainSubsquidHistoryOperationFactory(url: blockExplorer?.url)
-        case .alchemy, .etherscan:
+        case .alchemy, .etherscan, .reef:
```
Comment (DRadmir) [inline fearless/CoreLayer/OperationFactory/BlockExplorer/History/Staking/ParachainHistoryOperationFactory.swift:25]: why?

File: fearless/CoreLayer/OperationFactory/BlockExplorer/Rewards/ReefRewardOperationFactory.swift [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+import SoraFoundation
+import SSFModels
+
+enum ReefRewardOperationFactoryError: Error {
+    case urlMissing
+    case stakingTypeUnsupported
+    case incorrectAddress
+}
+
+final class ReefRewardOperationFactory {
```
Comment (DRadmir) [inline fearless/CoreLayer/OperationFactory/BlockExplorer/Rewards/ReefRewardOperationFactory.swift:13]: I guess this doesn't work?

File: fearless/Modules/NewWallet/WalletSendConfirm/WalletSendConfirmInteractor.swift [hunk 11]
```diff
@@ -76,6 +82,28 @@ final class WalletSendConfirmInteractor: RuntimeConstantFetching {
...
+            return items
+        }
+
+        return try await withUnsafeThrowingContinuation { continuation in
+            let runtimeItemsOperation = runtimeItemRepository.fetchAllOperation(with: RepositoryFetchOptions())
+
+            runtimeItemsOperation.completionBlock = { [weak self] in
+                do {
+                    let items = try runtimeItemsOperation.extractNoCancellableResultData()
+                    self?.runtimeItems = items
```
Comment (DRadmir) [inline fearless/Modules/NewWallet/WalletSendConfirm/WalletSendConfirmInteractor.swift:96]: what if save map?

File: fearless/Modules/Send/SendInteractor.swift [hunk 12]
```diff
@@ -84,9 +91,32 @@ final class SendInteractor: RuntimeConstantFetching {
...
+            return items
+        }
+
+        return try await withUnsafeThrowingContinuation { continuation in
+            let runtimeItemsOperation = runtimeItemRepository.fetchAllOperation(with: RepositoryFetchOptions())
+
+            runtimeItemsOperation.completionBlock = { [weak self] in
+                do {
+                    let items = try runtimeItemsOperation.extractNoCancellableResultData()
+                    self?.runtimeItems = items
```
Comment (DRadmir) [inline fearless/Modules/Send/SendInteractor.swift:105]: what if save map?

File: fearless/Modules/Staking/Operations/IdentityOperationFactory.swift [hunk 13]
```diff
@@ -105,13 +111,13 @@ final class IdentityOperationFactory {
     ) -> CompoundOperationWrapper<[AccountAddress: AccountIdentity]> {
         let path = StorageCodingPath.identity
 
-        let keyParams: () throws -> [Data] = {
+        let keyParams: () throws -> [AccountAddress] = {
```
Comment (DRadmir) [inline fearless/Modules/Staking/Operations/IdentityOperationFactory.swift:114]: Does successful work with 14 metadata?

File: fearless/Resources/chains.json
Comment (DRadmir) [inline fearless/Resources/chains.json:1]: discard changes?

**Review by DRadmir — CHANGES_REQUESTED** [review]

File: fearless/Common/Network/BlockExplorer/Subsquid/ReefSubsquidHistory.swift [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+    let transfers: [GiantsquidTransfer]?
+
+    var history: [WalletRemoteHistoryItemProtocol] {
+        transfers ?? []
+    }
+}
+
+extension ReefResponseData: RewardOrSlashResponse {
+    var data: [RewardOrSlashData] {
+        []
```
[the remark this answers is not in the document]
Comment (bnsports) [reply fearless/Common/Network/BlockExplorer/Subsquid/ReefSubsquidHistory.swift:70]: not implemented yet

**Review by DRadmir — APPROVED** [review]

**Review by DRadmir — DISMISSED** [review]

**Review by DRadmir — APPROVED** [review]

**Pull request merged**